ajax/downVoteAjax.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+include('config/config.php');
+include('config/init.php');
+
+//MAKE IT SO IT CAN ONLY DO THIS ONCE!
+//right now it can do it as many times as it wants
+downVote($_REQUEST['eventID']);
+
+$holder= $_REQUEST['eventVotes'];
```

Comment:
This seems a bit weird to me. You're letting the user pass you the number of current votes and then decrementing it. Instead. If you trust the client to be right, why not just decrement it in the client and leave the server out of it? If you don't trust the client, you'd want to actually get this value from the db rather than taking it as a request parameter.

Reply:
I thought that the request in this case was coming from the js function (which has eventID and eventVotes passed to it?), which is coming from the client, not the user?

Reply:
It's coming from the client, but that is "the user" as far as this endpoint is concerned. My point is that the client is passing the vote total to the server, but the server already knows the vote total because it has access to the database.